src/lib/coda_base/user_command_status.ml

```diff
@@ -125,6 +144,537 @@ module Failure = struct
+        ~value_to_hlist:Poly.to_hlist ~value_of_hlist:Poly.of_hlist
+        ~var_to_hlist:Poly.to_hlist ~var_of_hlist:Poly.of_hlist
+
+    let none =
+      { predicate= false
+      ; source_not_present= false
+      ; receiver_not_present= false
+      ; amount_insufficient_to_create_account= false
+      ; cannot_pay_creation_fee_in_token= false
+      ; source_insufficient_balance= false
+      ; source_minimum_balance_violation= false
+      ; receiver_already_exists= false
+      ; not_token_owner= false
+      ; mismatched_token_permissions= false
+      ; overflow= false
+      ; signed_command_on_snapp_account= false
+      ; snapp_account_not_present= false
+      ; update_not_permitted= false
+      ; incorrect_nonce= false }
+
+    let predicate = {none with predicate= true}
+
+    let source_not_present = {none with source_not_present= true}
+
+    let receiver_not_present = {none with receiver_not_present= true}
+
+    let amount_insufficient_to_create_account =
+      {none with amount_insufficient_to_create_account= true}
+
+    let cannot_pay_creation_fee_in_token =
+      {none with cannot_pay_creation_fee_in_token= true}
+
+    let source_insufficient_balance =
+      {none with source_insufficient_balance= true}
+
+    let source_minimum_balance_violation =
+      {none with source_minimum_balance_violation= true}
+
+    let receiver_already_exists = {none with receiver_already_exists= true}
+
+    let not_token_owner = {none with not_token_owner= true}
+
+    let mismatched_token_permissions =
+      {none with mismatched_token_permissions= true}
+
+    let overflow = {none with overflow= true}
+
+    let signed_command_on_snapp_account =
+      {none with signed_command_on_snapp_account= true}
+
+    let snapp_account_not_present = {none with snapp_account_not_present= true}
+
+    let update_not_permitted = {none with update_not_permitted= true}
+
+    let incorrect_nonce = {none with incorrect_nonce= true}
+
+    let to_enum = function
+      | {predicate= true; _} ->
+          1
+      | {source_not_present= true; _} ->
+          2
+      | {receiver_not_present= true; _} ->
+          3
+      | {amount_insufficient_to_create_account= true; _} ->
+          4
+      | {cannot_pay_creation_fee_in_token= true; _} ->
+          5
+      | {source_insufficient_balance= true; _} ->
+          6
+      | {source_minimum_balance_violation= true; _} ->
+          7
+      | {receiver_already_exists= true; _} ->
+          8
+      | {not_token_owner= true; _} ->
+          9
+      | {mismatched_token_permissions= true; _} ->
+          10
+      | {overflow= true; _} ->
+          11
+      | {signed_command_on_snapp_account= true; _} ->
+          12
+      | {snapp_account_not_present= true; _} ->
+          13
+      | {update_not_permitted= true; _} ->
+          14
+      | {incorrect_nonce= true; _} ->
+          15
+      | _ ->
+          0
+
+    let of_enum = function
+      | 0 ->
```

Comment:
would it make sense to match on `of_enum n`, and if that's, say, `Predicate`, return `Some predicate` ?